api/api.go

```diff
@@ -83,12 +83,16 @@ func NewAPIWithVersion(ctx context.Context, globalConfig *conf.GlobalConfigurati
 		}
 
 		r.Get("/settings", api.Settings)
+
 		r.Get("/authorize", api.ExternalProviderRedirect)
-		r.Post("/signup", api.Signup)
+
 		r.With(api.requireAdminCredentials).Post("/invite", api.Invite)
-		r.Post("/recover", api.Recover)
-		r.Post("/verify", api.Verify)
-		r.Post("/token", api.Token)
+
+		r.With(api.requireEmailProvider).Post("/signup", api.Signup)
+		r.With(api.requireEmailProvider).Post("/recover", api.Recover)
+		r.With(api.requireEmailProvider).Post("/verify", api.Verify)
```

Comment:
In the rare case that an external provider's emails are not verified, they need to able to access this endpoint.

Reply:
good call, I'll fix that.